contracts/orbital-core/src/contract.rs

```diff
@@ -1,119 +1,171 @@
+use crate::{
+    admin_logic::admin,
+    state::{OrbitalDomainConfig, UserConfig, USER_NONCE},
+    user_logic::user,
+    utils::{extract_ica_identifier_from_port, get_ica_identifier, OpenAckVersion},
+};
 #[cfg(not(feature = "library"))]
 use cosmwasm_std::entry_point;
 use cw2::set_contract_version;
-use cw_ownable::{
-    assert_owner, get_ownership, initialize_owner, update_ownership, Action, OwnershipError,
+use cw_ownable::{get_ownership, initialize_owner};
+use neutron_sdk::{
+    bindings::{msg::NeutronMsg, query::NeutronQuery},
+    sudo::msg::SudoMsg,
+    NeutronResult,
 };
 
 use crate::{
-    account_types::UncheckedOrbitalDomainConfig,
-    error::ContractError,
-    msg::{ExecuteMsg, InstantiateMsg, QueryMsg},
-    state::{UserConfig, ORBITAL_DOMAINS, USER_CONFIGS},
+    msg::{ExecuteMsg, InstantiateMsg, MigrateMsg, QueryMsg},
+    state::{CLEARING_ACCOUNTS, ORBITAL_DOMAINS, USER_CONFIGS},
 };
 use cosmwasm_std::{
-    to_json_binary, Addr, Binary, BlockInfo, Deps, DepsMut, Env, MessageInfo, Response, StdResult,
+    to_json_binary, Addr, Binary, Deps, DepsMut, Env, MessageInfo, Reply, Response, StdError,
+    StdResult, Uint64,
 };
 
 pub const CONTRACT_NAME: &str = "orbital-core";
 pub const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
 
+pub type OrbitalResult = NeutronResult<Response<NeutronMsg>>;
+pub type QueryDeps<'a> = Deps<'a, NeutronQuery>;
+pub type ExecuteDeps<'a> = DepsMut<'a, NeutronQuery>;
+
 #[entry_point]
 pub fn instantiate(
-    deps: DepsMut,
+    deps: ExecuteDeps,
     _env: Env,
     _info: MessageInfo,
     msg: InstantiateMsg,
-) -> Result<Response, OwnershipError> {
+) -> OrbitalResult {
     set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
     initialize_owner(deps.storage, deps.api, Some(&msg.owner))?;
 
+    USER_NONCE.save(deps.storage, &Uint64::zero())?;
     Ok(Response::new())
 }
 
 #[entry_point]
 pub fn execute(
-    deps: DepsMut,
+    deps: ExecuteDeps,
     env: Env,
     info: MessageInfo,
     msg: ExecuteMsg,
-) -> Result<Response, ContractError> {
+) -> NeutronResult<Response<NeutronMsg>> {
     match msg {
+        // admin action to manage ownership of orbital-core
```

Comment:
Not really an admin action because accepting ownership is done by the user, so maybe like mentioned above, move it out of admin module and rename?

Reply:
imho it's still an admin action from a domain-driven perspective. did the renaming though, good point!

Reply:
Yea most actions are admin, I personally separate the update_ownership in its own category but here it's also fine